src/lib/install.sh

```diff
@@ -7,18 +7,61 @@ _install_name_='install.sh'
 
 install_single() {
     #: Used to install single-binary recipes
-    #: $1 - path to source, $2 - name, $3 - version, $4 (optional) - relative path to binary (w/o the first dir)
-    dir_fullpath="$_LOCAL/opt/$2/$3"
-    mkdir -p "$dir_fullpath"
-    ln -fs  "$dir_fullpath" "$_LOCAL/opt/$2/current"
-    cp -pr "$1" "$dir_fullpath/$2"
-    if [ -z "$4" ]; then
-        path_to_bin="$2"
+    #:    ($1) source_path - path to the source directory/file
+    #:    ($2) recipe_name - the name of the recipe currently being installed
+    #:    ($3) version - the version of the recipe currently being installed
+    #:    ($4) bin_path [optional] - in case of archive - relative path to binary inside
+    local source_path="$1"; local recipe_name="$2"; local version="$3"; local bin_path="${4:-$recipe_name}"
```

Comment:
Let us have these one per line -- would help both for better readability and easier changes (like adding/removing an argument).

Reply:
Maybe also move the documenting comments on the actual argument line as well, like:
```bash
  local recipe_name="$2"  #: recipe_name - the name of the recipe currently being installed
```